src/cdk/a11y/key-manager/tree-key-manager.ts

```diff
@@ -0,0 +1,417 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken, QueryList} from '@angular/core';
+import {coerceObservable} from '@angular/cdk/coercion/private';
+import {Observable, Subject, Subscription, isObservable, of as observableOf} from 'rxjs';
+import {take} from 'rxjs/operators';
+import {
+  TreeKeyManagerFactory,
+  TreeKeyManagerItem,
+  TreeKeyManagerOptions,
+  TreeKeyManagerStrategy,
+} from './tree-key-manager-strategy';
+import {Typeahead} from './typeahead';
+
+/**
+ * This class manages keyboard events for trees. If you pass it a QueryList or other list of tree
+ * items, it will set the active item, focus, handle expansion and typeahead correctly when
+ * keyboard events occur.
+ */
+export class TreeKeyManager<T extends TreeKeyManagerItem> implements TreeKeyManagerStrategy<T> {
+  /** The index of the currently active (focused) item. */
+  private _activeItemIndex = -1;
+  /** The currently active (focused) item. */
+  private _activeItem: T | null = null;
+  /** Whether or not we activate the item when it's focused. */
+  private _shouldActivationFollowFocus = false;
+  /**
+   * The orientation that the tree is laid out in. In `rtl` mode, the behavior of Left and
+   * Right arrow are switched.
+   */
+  private _horizontalOrientation: 'ltr' | 'rtl' = 'ltr';
+
+  // Keep tree items focusable when disabled. Align with
```

Comment:
It's not immediately clear what this comment is trying to say. Is it part of the docs for `_skipPredicateFn`?